### PR TITLE
[SPARK-47218] [SQL] XML: Changed SchemaOfXml to fail on DROPMALFORMED mode

### PR DESCRIPTION
**Pull request by yhosny**
### What changes were proposed in this pull request?

Changed schema_of_xml should fail with an error on DROPMALFORMED mode to avoid creating schemas out of invalid XML.

### Why are the changes needed?

DROPMALFORMED parse mode imply silently dropping the malformed record. But SchemaOfXml is expected to return a schema and may not have a valid schema to return for a malformed record. So DROPMALFORMED cannot be supported..

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Unit test.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by sandip-db** [comment]
Why are the changes needed?

DROPMALFORMED parse mode imply silently dropping the malformed record. But SchemaOfXml is expected to return a schema and may not have a valid schema to return for a malformed record. So DROPMALFORMED cannot be supported..

**Review by sandip-db — APPROVED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/XmlSuite.scala [hunk 1]
```diff
@@ -1302,6 +1302,22 @@ class XmlSuite
     assert(result.select("decoded._corrupt_record").head().getString(0).nonEmpty)
   }
 
+  test("schema_of_xml parse error test") {
+    val e = intercept[AnalysisException] {
+       spark.sql(s"""SELECT schema_of_xml('<ROW><a>1<ROW>', map('mode', 'DROPMALFORMED'))""")
```
Comment (sandip-db) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/XmlSuite.scala:1307]: can you please add permissive and failfast mode scenarios as well in this test? Thanks.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by HyukjinKwon** [comment]
(let's file a new JIRA next time)

**Pull request closed without merging**